lib/spack/spack/build_environment.py

```diff
@@ -232,6 +232,17 @@ def clean_environment():
         # show useful matches.
         env.set('LC_ALL', build_lang)
 
+    remove_flags = set()
+    keep_flags = set()
+    if spack.config.get('config:flags:keep_werror') == 'all':
+        keep_flags.add('-Werror*')
+    else:
+        if spack.config.get('config:flags:keep_werror') == 'specific':
+            keep_flags.add('-Werror=*')
+        remove_flags.add('-Werror*')
+    env.set('SPACK_COMPILER_FLAGS_KEEP', ' | '.join(keep_flags))
```

Comment:
I didn't realize whitespace was allowed between alternatives in a case-match expression! I guess we don't need to add any explanation of that here, but if we're setting this env var *without* spaces between alternatives in `SPACK_COMPILER_FLAGS_REMOVE='-Werror*|-llib1|-Wl*'` in the testing added in this PR, I think it would be a very good idea to do the same here and avoid spaces, or to add spaces in our test cases.

Reply:
In the end I think it works the same either way, I'll test it and maybe remove the spacing.

Reply:
Removed.